common/perf-test-core/src/main/java/com/azure/perf/test/core/MockEventProcessorTest.java

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.perf.test.core;
+
+import com.beust.jcommander.Parameter;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.function.Consumer;
+
+/**
+ * Represents a Mock Event Processor Test.
+ */
+public class MockEventProcessorTest extends EventPerfTest<MockEventProcessorTest.MockEventProcessorPerfOptions> {
+    private final MockEventProcessor mockEventProcessor;
+
+    /**
+     * Creates an instance of Mock Event Processor Test
+     * @param perfStressOptions the options to used to configure the test.
+     */
+    public MockEventProcessorTest(MockEventProcessorPerfOptions perfStressOptions) {
+        super(perfStressOptions);
+        Consumer<MockErrorContext> errorprocessor = mockErrorContext -> errorRaised();
+
+        Consumer<MockEventContext> eventPrcessor = mockEventContext -> eventRaised();
+
+        Duration errorAfter = perfStressOptions.getErrorAfterInSeconds() > 0
+            ? Duration.ofSeconds(perfStressOptions.getErrorAfterInSeconds()) : null;
+
+        mockEventProcessor = new MockEventProcessor(2, perfStressOptions.getMaxEventsPerSecond(), errorAfter,
+            errorprocessor, eventPrcessor);
+    }
+
+    @Override
+    public Mono<Void> setupAsync() {
+        return super.setupAsync().then(Mono.defer(() -> {
+            mockEventProcessor.start();
+            return Mono.empty();
+        }));
+    }
+
+    @Override
+    public Mono<Void> cleanupAsync() {
+        return Mono.defer(() -> {
+            mockEventProcessor.stop();
+            return Mono.empty();
+        }).then(super.cleanupAsync());
+    }
+
+    /**
+     * Represents the perf options for Mock Event Processor Test.
+     */
+    public static class MockEventProcessorPerfOptions extends PerfStressOptions {
```

Comment:
Add parameter `int partitions`.

https://github.com/Azure/azure-sdk-for-net/blob/main/common/Perf/Azure.Sample.Perf/Event/MockEventProcessorOptions.cs#L17-L18

Reply:
added